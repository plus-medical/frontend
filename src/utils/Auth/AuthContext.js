import React, { useState, createContext, useEffect } from 'react'
import { useAuth } from './useAuth'

const AuthContext = createContext()
const NAME_TOKEN = 'token'

function AuthProvider ({ children }) {
  const auth = useAuth()
  const [authenticated, setAuthenticated] = useState(true)
  const { data, setData, error, logout } = auth

  const handleLogin = (params) => {
    auth.signIn(params)
  }

  const handleLogout = () => {
    logout()
    window.localStorage.removeItem(NAME_TOKEN)
    window.localStorage.removeItem('role')
  }

  const handleSignUp = async (data) => {
    return await auth.signUp(data)
  }

  useEffect(() => {
    const token = window.localStorage.getItem(NAME_TOKEN)
    if (token) {
      setAuthenticated(true)
    } else if (error === '' && data.user) {
      setAuthenticated(true)
      !token && window.localStorage.setItem(NAME_TOKEN, true)
      window.localStorage.setItem('role', data.user.role)
    } else {
      setAuthenticated(false)
      window.localStorage.removeItem(NAME_TOKEN)
    }
  }, [JSON.stringify(data), error])

  return (
    <AuthContext.Provider
      value={{
        authenticated,
        error,
        data,
        setAuthenticated,
        handleLogin,
        handleLogout,
        handleSignUp
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export { AuthContext, AuthProvider }
