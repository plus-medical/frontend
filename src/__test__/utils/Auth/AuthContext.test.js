/* eslint-disable no-undef */
import React from 'react'
import { create } from 'react-test-renderer'
import RouterMock from '../../../__mocks__/RoutesMocks'
import AuthContext from '../../../utils/Auth/AuthContext'

describe('<AuthContext />', () => {
  test('Review Snapshot in AuthContext component', () => {
    const auth = create(
      <RouterMock>
        <AuthContext />
      </RouterMock>
    )
    expect(auth.toJSON()).toMatchSnapshot()
  })
})
