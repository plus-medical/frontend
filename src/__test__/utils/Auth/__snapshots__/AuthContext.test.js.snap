// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthContext /> Review Snapshot in AuthContext component 1`] = `null`;
